core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/TransformCorrelatedGlobalAggregationWithProjection.java

```diff
@@ -287,4 +306,21 @@ public Result apply(CorrelatedJoinNode correlatedJoinNode, Captures captures, Co
                 globalAggregation,
                 assignments));
     }
+
+    private static boolean isNullMaskNeeded(AggregationNode node)
+    {
+        return getBooleanAggregationSymbol(node).isEmpty();
+    }
+
+    private static Optional<Symbol> getBooleanAggregationSymbol(AggregationNode node)
+    {
+        if (node.getAggregations().size() != 1) {
+            return Optional.empty();
+        }
+        Aggregation aggregation = getOnlyElement(node.getAggregations().values());
+        if (aggregation.getResolvedFunction().name().equals(BOOL_OR) && aggregation.getArguments().getFirst() instanceof Reference symbolReference) {
```

Comment:
what about mask and filter (should be empty)

Reply:
i will add check for that bit right now there is no possibility to have suche case